src/entities.lisp

```diff
@@ -0,0 +1,96 @@
+;;;; entities.lisp
+
+(in-package #:sketch)
+
+;;;  _____ _   _ _____ ___ _____ ___ _____ ____
+;;; | ____| \ | |_   _|_ _|_   _|_ _| ____/ ___|
+;;; |  _| |  \| | | |  | |  | |  | ||  _| \___ \
+;;; | |___| |\  | | |  | |  | |  | || |___ ___) |
+;;; |_____|_| \_| |_| |___| |_| |___|_____|____/
+;;;
+
+;;; TODO:
+;;; - Better code reuse with sketch.lisp
+
+(defparameter *entity* nil
+  "The current entity.")
+
+(defclass entity ()
+  ((width :initform 100 :accessor entity-width :initarg :width)
+   (height :initform 100 :accessor entity-height :initarg :height)))
+
+(defmethod register-entity ((sketch sketch) (entity entity) box)
```

Comment:
Would it make sense to have a tree of entities? In this case you'd want to register an entity to its parent instead of a sketch, which also could make propagation cheaper.

Reply:
Oh, definitely. I also want them to live in a quadtree, but that's for later.